spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMetadataTables.java

```diff
@@ -319,6 +319,111 @@ public void testAllFilesPartitioned() throws Exception {
+    Assert.assertEquals("testBranch", testBranch.get(0).getAs("name"));
+    Assert.assertEquals("BRANCH", testBranch.get(0).getAs("type"));
+    Assert.assertEquals(currentSnapshotId, testBranch.get(0).getAs("snapshot_id"));
+    Assert.assertEquals(Long.valueOf(10), testBranch.get(0).getAs("max_reference_age_in_ms"));
+    Assert.assertEquals(Integer.valueOf(20), testBranch.get(0).getAs("min_snapshots_to_keep"));
+    Assert.assertEquals(Long.valueOf(30), testBranch.get(0).getAs("max_snapshot_age_in_ms"));
+
+    // Check tag entries in refs table
+    List<Row> testTag =
+        spark.sql("SELECT * FROM " + tableName + ".refs WHERE name = 'testTag' AND type='TAG'").collectAsList();
+    Assert.assertEquals("testTag", testTag.get(0).getAs("name"));
+    Assert.assertEquals("TAG", testTag.get(0).getAs("type"));
+    Assert.assertEquals(currentSnapshotId, testTag.get(0).getAs("snapshot_id"));
+    Assert.assertEquals(Long.valueOf(50), testTag.get(0).getAs("max_reference_age_in_ms"));
+
+    // Check projection in refs table
+    List<Row> testTagProjection =
+        spark.sql("SELECT name,type,snapshot_id,max_reference_age_in_ms,min_snapshots_to_keep FROM " + tableName +
+            ".refs where type='TAG'").collectAsList();
+    Assert.assertEquals("testTag", testTagProjection.get(0).getAs("name"));
+    Assert.assertEquals("TAG", testTagProjection.get(0).getAs("type"));
+    Assert.assertEquals(currentSnapshotId, testTagProjection.get(0).getAs("snapshot_id"));
+    Assert.assertEquals(Long.valueOf(50), testTagProjection.get(0).getAs("max_reference_age_in_ms"));
+    Assert.assertNull(testTagProjection.get(0).getAs("min_snapshots_to_keep"));
+
+    List<Row> mainBranchProjection =
+        spark.sql("SELECT name,type FROM " + tableName + ".refs WHERE name = 'main' AND type='BRANCH'").collectAsList();
+    Assert.assertEquals("main", mainBranchProjection.get(0).getAs("name"));
+    Assert.assertEquals("BRANCH", mainBranchProjection.get(0).getAs("type"));
+    Assert.assertThrows(IllegalArgumentException.class, () -> mainBranchProjection.get(0).getAs("snapshot_id"));
+
+    List<Row> testBranchProjection =
+        spark.sql("SELECT type,name,max_reference_age_in_ms,snapshot_id FROM " + tableName +
+            ".refs WHERE name = 'testBranch' AND type='BRANCH'").collectAsList();
+    Assert.assertEquals("testBranch", testBranchProjection.get(0).getAs("name"));
+    Assert.assertEquals("BRANCH", testBranchProjection.get(0).getAs("type"));
+    Assert.assertEquals(currentSnapshotId, testBranchProjection.get(0).getAs("snapshot_id"));
+    Assert.assertEquals(Long.valueOf(10), testBranchProjection.get(0).getAs("max_reference_age_in_ms"));
+    Assert.assertThrows(
+        IllegalArgumentException.class,
+        () -> testBranchProjection.get(0).getAs("min_snapshots_to_keep"));
+    Assert.assertThrows(
+        IllegalArgumentException.class,
+        () -> testBranchProjection.get(0).getAs("max_snapshot_age_in_ms"));
```

Comment:
[minor] we can skip these assertions, IMHO we can assume engines project operator would handle it, and only returns columns being projected.